tests/e2e/monitoring/test_monitoring_on_negative_scenarios.py

```diff
@@ -552,3 +583,33 @@ def test_monitoring_delete_pvc(self):
         assert prometheus_health_check(), (
             "Prometheus cluster health is not OK"
         )
+
+    @pytest.mark.polarion_id("OCS-636")
+    def test_monitoring_shut_prometheus_node(self, delete_configmap, pods, nodes):
+        """
+        Test case to validate whether shutdown and recovery of a
+        node where monitoring pods running has no functional impact
+
+        """
+        # Get all prometheus pods
+        prometheus_pod_obj_list = pod.get_all_pods(
+            namespace=defaults.OCS_MONITORING_NAMESPACE, selector=['prometheus']
+        )
+
+        # Get the node where the prometheus pod is hosted
+        prometheus_node_obj = pod.get_pod_node(prometheus_pod_obj_list[0])
+
+        # Shutdown node where the prometheus pod is hosted
+        nodes.stop_nodes(nodes=[prometheus_node_obj])
+
+        log.info('Waiting for 20 seconds')
+        time.sleep(20)
+
+        # Powering on node
+        nodes.start_nodes(nodes=[prometheus_node_obj])
+
+        # check pvcdata collected on prometheus
+        for pod_obj in pods:
+            assert check_pvcdata_collected_on_prometheus(pod_obj.pvc.name), (
```

Comment:
I would rather iterated over all pods using log.error() instead of assert, and then after the for loop, used assert to check if at least one pod failed the check. This way, we will have more evidence in the logs (how many pods were checked? how many of these failed? ...) which will be helpful during evaluation of the results.